ocpi-toolkit-2.2.1/src/main/kotlin/com/izivia/ocpi/toolkit/common/HttpUtils.kt

```diff
@@ -70,45 +71,53 @@ fun String.decodeBase64(): String = Base64.getDecoder().decode(this).decodeToStr
 fun authorizationHeader(token: String): Pair<String, String> = Header.AUTHORIZATION to "Token ${token.encodeBase64()}"
 
 /**
- * Creates the authorization header by taking the right token in the platform repository
+ * Creates the authorization header by taking the client token (or the token A if allowed) in the platform repository
+ *
+ * @receiver PlatformRepository used to retrieve tokens
+ * @param platformUrl partner /versions url
+ * @param allowTokenA only true when called on versions / credentials module
+ * @return Pair<String, String>
  */
-suspend fun PlatformRepository.buildAuthorizationHeader(baseUrl: String, allowTokenAOrTokenB: Boolean = false) =
-    if (allowTokenAOrTokenB) {
-        getCredentialsTokenC(platformUrl = baseUrl)
-            ?: getCredentialsTokenB(platformUrl = baseUrl)
-            ?: getCredentialsTokenA(platformUrl = baseUrl)
-            ?: throw throw OcpiClientGenericException(
-                "Could not find CREDENTIALS TOKEN A OR B OR C associated with platform $baseUrl"
+suspend fun PlatformRepository.buildAuthorizationHeader(
+    platformUrl: String,
+    allowTokenA: Boolean = false
```

Comment:
I wonder if  we want to "allow" the use of tokenA, or actually force to use it? The only case of using tokenA is when we initialize the credentials handshake, where we call 3 endpoints with tokenA (and in that case we are sure we want to use tokenA). Then we will always use the client token. I will go on with the review to see if what I say makes sense or not.

Reply:
created issue: https://github.com/IZIVIA/ocpi-toolkit/issues/19